public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistory.tsx

```diff
@@ -0,0 +1,368 @@
+import { css, cx } from '@emotion/css';
+import React, { useCallback, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useMeasure } from 'react-use';
+
+import { GrafanaTheme2, TimeRange } from '@grafana/data';
+import { isFetchError } from '@grafana/runtime';
+import { SceneComponentProps, SceneObjectBase, sceneGraph } from '@grafana/scenes';
+import {
+  Alert,
+  Button,
+  Field,
+  Icon,
+  Input,
+  Label,
+  LoadingBar,
+  Stack,
+  Text,
+  Tooltip,
+  useStyles2,
+  withErrorBoundary,
+} from '@grafana/ui';
+import { EntityNotFound } from 'app/core/components/PageNotFound/EntityNotFound';
+import { Trans, t } from 'app/core/internationalization';
+import {
+  GrafanaAlertStateWithReason,
+  getReasonFromState,
+  isAlertStateWithReason,
+  isGrafanaAlertState,
+  mapStateWithReasonToBaseState,
+} from 'app/types/unified-alerting-dto';
+
+import { stateHistoryApi } from '../../../api/stateHistoryApi';
+import { GRAFANA_DATASOURCE_NAME } from '../../../utils/datasource';
+import { stringifyErrorLike } from '../../../utils/misc';
+import { AlertLabels } from '../../AlertLabels';
+import { CollapseToggle } from '../../CollapseToggle';
+import { STATE_HISTORY_POLLING_INTERVAL } from '../state-history/LokiStateHistory';
+import { LogRecord } from '../state-history/common';
+import { useRuleHistoryRecords } from '../state-history/useRuleHistoryRecords';
+
+const LIMIT_LABELS = 5;
+const LIMIT_EVENTS = 250;
+
+const HistoryEventsList = ({ timeRange }: { timeRange?: TimeRange }) => {
+  const styles = useStyles2(getStyles);
+  const { useGetRuleHistoryQuery } = stateHistoryApi;
+  // Filter state
+  const [eventsFilter, setEventsFilter] = useState('');
+  // form for filter fields
+  const { setValue, register, handleSubmit } = useForm({ defaultValues: { query: '' } }); //  form for search field
+  const from = timeRange?.from.unix();
+  const to = timeRange?.to.unix();
+  const onFilterCleared = useCallback(() => {
+    setEventsFilter('');
+    setValue('query', '');
+  }, [setEventsFilter, setValue]);
+
+  const {
+    currentData: stateHistory,
+    isLoading,
+    isError,
+    error,
+  } = useGetRuleHistoryQuery(
+    {
+      from: from,
+      to: to,
+      limit: LIMIT_EVENTS,
+    },
+    {
+      refetchOnFocus: true,
+      refetchOnReconnect: true,
+      pollingInterval: STATE_HISTORY_POLLING_INTERVAL,
```

Comment:
Would this conflict with the refresh interval from the scenes?

Reply:
good point, let's split it